secure/key-manager/src/tests.rs

```diff
@@ -471,10 +497,39 @@ fn verify_manual_consensus_rotation<T: LibraInterface>(config: NodeConfig, mut n
     assert_ne!(new_pubkey, genesis_pubkey);
     assert_eq!(new_pubkey, new_config.consensus_pubkey);
     assert_eq!(&new_pubkey, new_info.consensus_public_key());
+
+    // Check secure storage not updated as key rotation was only on-chain
+    assert_ne!(
+        new_pubkey,
+        node.key_manager
+            .storage
+            .get_public_key(crate::CONSENSUS_KEY)
+            .unwrap()
+            .public_key
+    );
+
+    // Update secure storage manually and verify
+    assert!(node
+        .key_manager
+        .storage
+        .set(crate::CONSENSUS_KEY, Value::Ed25519PrivateKey(new_privkey))
+        .is_ok());
+    assert_eq!(
+        new_pubkey,
+        node.key_manager
+            .storage
+            .get_public_key(crate::CONSENSUS_KEY)
+            .unwrap()
+            .public_key
+    );
+    assert_eq!(
+        Action::NoAction,
+        node.key_manager.evaluate_status().unwrap()
+    );
```

Comment:
I think what we want, if we want to do this is to generate the key in secure storage. Secure storage should be updated prior to on chain rotation. Though this was never intended to be part of this test.

Reply:
Aah, I see. The test name confused me; I thought it was meant to be a manual rotation for the entire pipeline. I've updated the test name and removed the storage overwrite logic, so this test is essentially the same now.